cardinal/query.go

```diff
@@ -0,0 +1,59 @@
+package cardinal
+
+import (
+	"pkg.world.dev/world-engine/cardinal/ecs"
+	"pkg.world.dev/world-engine/cardinal/ecs/filter"
+	"pkg.world.dev/world-engine/cardinal/ecs/storage"
+)
+
+// Query allowed for the querying of entities within a World.
+type Query struct {
+	impl *ecs.Query
+}
+
+// NewQuery creates a new Query.
+func NewQuery(filter LayoutFilter) *Query {
+	return &Query{ecs.NewQuery(filter)}
+}
+
+// QueryCallBackFn represents a function that can operate on a single EntityID.
+type QueryCallBackFn func(EntityID) bool
+
+// Each executes the given callback function on every EntityID that matches this query.
+func (q *Query) Each(w *World, callback QueryCallBackFn) {
+	q.impl.Each(w.impl, func(eid storage.EntityID) bool {
+		return callback(eid)
+	})
+}
```

Comment:
we should mention the behavior of returning true/false in the callback function in the comment

Reply:
Added comment.